src/log_batch.rs

```diff
@@ -1531,6 +1531,49 @@ mod tests {
         ));
     }
 
+    #[test]
+    fn test_corruption() {
+        let region_id = 7;
+        let data = vec![b'x'; 16];
+        let mut batch = LogBatch::default();
+        batch
+            .add_entries::<Entry>(region_id, &generate_entries(1, 11, Some(&data)))
+            .unwrap();
+        batch
+            .put(region_id, b"key".to_vec(), b"value".to_vec())
+            .unwrap();
+        // enable compression so that len_and_type > len.
+        batch.finish_populate(1).unwrap();
+        let file_context = LogFileContext::new(FileId::dummy(LogQueue::Append), Version::default());
+        batch.prepare_write(&file_context).unwrap();
+        let encoded = batch.encoded_bytes();
```

Comment:
Still missing a valid decode so we can verify it won't panic anymore.

Reply:
There're plenty of tests above testing the valid cases.